docs/changes/wcs/16411.bug.rst

```diff
@@ -0,0 +1 @@
+Fixed an issue which caused calls to WCS coordinate conversion routines to not be thread-safe due to repeated calls to WCS.wcs.set()
```

Comment:
Might be good to clarify that memory race was occurring because *each* thread was calling `WCS.wcs.set()` in parallel, and not because of repeated calls to `wcsset()`.